components/form/NameNsDescription.vue

```diff
@@ -136,7 +136,7 @@ export default {
     },
 
     description(val) {
-      this.value.setAnnotation(DESCRIPTION, val);
+      this.value.metadata[DESCRIPTION] = val;
```

Comment:
Why?  (setAnnotation also removes the annotation if the value is empty)

Reply:
Switched back. I did this because the value didn't have setAnnotation. 

It turns out that the existing use of NameNSDescription was passing metadata to value and not the resource value. For whatever reason that didn't prevent the name from getting set which I haven't spent the time to figure out why.